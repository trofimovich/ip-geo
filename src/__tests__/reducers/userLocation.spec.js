import userLocationReducer from '../../reducers/userLocation';

describe('locationReducer', () => {
  it('should return initial state', () => {
    expect(userLocationReducer(undefined, {})).toMatchSnapshot();
  });

  it('should handle "LOCATION_USER_REQUEST_INIT"', () => {
    const reduced = userLocationReducer({
      isFetching: false,
      location: null,
    }, { type: 'LOCATION_USER_REQUEST_INIT' });

    expect(reduced.isFetching).toBe(true);
    expect(reduced.location).toBeNull();
  });

  it('should handle "LOCATION_USER_REQUEST_SUCCESS"', () => {
    const reduced = userLocationReducer({
      isFetching: true,
      location: null,
    }, { type: 'LOCATION_USER_REQUEST_SUCCESS', location: { country: 'UK' } });

    expect(reduced.isFetching).toBe(false);
    expect(reduced.location).toEqual({ country: 'UK' });
  });

  it('should handle "LOCATION_USER_REQUEST_ERROR"', () => {
    const reduced = userLocationReducer({
      isFetching: true,
      location: null,
    }, { type: 'LOCATION_USER_REQUEST_ERROR', error: 'Something went wrong.' });

    expect(reduced.isFetching).toBe(false);
    expect(reduced.location).toBeNull();
  });
});

