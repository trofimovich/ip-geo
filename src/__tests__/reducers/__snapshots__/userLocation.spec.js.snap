// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`locationReducer should return initial state 1`] = `
Object {
  "isFetching": false,
  "location": null,
}
`;
